File: 0x04-pointers_arrays_strings/5-rev_string.c
#include "holberton.h"

/**
 * rev_string - Reverse given string in place
 * @s: string to reverse
 */

void rev_string(char *s)
{
	char tmp;
	int i, j;

	for (i = 0; s[i] != '\0'; i++)
		;
	i--;
	for (j = 0; j <= i / 2; j++)
	{
		tmp = s[j];
		s[j] = s[i];
		s[i] = tmp;
	}
}

File: 0x04-pointers_arrays_strings/5-rev_string.go
//Reverse a string
package main

import "fmt"

func main() {
	var s string = "Holberton f̅ bròwn School"
	fmt.Println(s)
	rev_string(&s)
	fmt.Println(s)
}

// This doesn't preserve the position of the overdash
func rev_string(s *string) {
	r := make([]rune, len(*s))
	for i, c := range *s {
		r[len(*s) - 1 - i] = c
	}
	*s = string(r)
}

// Another option, still doesn't preserve all unicode
func rev_string2(s *string) {
	r := []rune(*s)
	for i, j := 0, len(r) - 1; i < len(r)/2; i , j = i + 1, j - 1 {
		r[i], r[j] = r[j], r[i]
	}
	*s = string(r)
}
